snowpack/src/commands/build.ts

```diff
@@ -117,602 +69,216 @@ async function installOptimizedDependencies(
-          (!isBundling || !/(?<!module)\.css$/.test(resolvedImportUrl));
-        const isAbsoluteUrlPath = path.posix.isAbsolute(resolvedImportUrl);
-        let resolvedImportPath = removeLeadingSlash(path.normalize(resolvedImportUrl));
-        // We treat ".proxy.js" files special: we need to make sure that they exist on disk
-        // in the final build, so we mark them to be written to disk at the next step.
-        if (isProxyImport) {
-          if (isAbsoluteUrlPath) {
-            this.filesToProxy.push(path.resolve(this.config.buildOptions.out, resolvedImportPath));
-          } else {
-            this.filesToProxy.push(path.resolve(path.dirname(outLoc), resolvedImportPath));
-          }
-
-          resolvedImportPath = resolvedImportPath + '.proxy.js';
-          resolvedImportUrl = resolvedImportUrl + '.proxy.js';
-        }
-
-        // When dealing with an absolute import path, we need to honor the baseUrl
-        if (isAbsoluteUrlPath) {
-          resolvedImportUrl = relativeURL(
-            path.dirname(outLoc),
-            path.resolve(this.config.buildOptions.out, resolvedImportPath),
-          );
-        }
-        // Make sure that a relative URL always starts with "./"
-        if (!resolvedImportUrl.startsWith('.') && !resolvedImportUrl.startsWith('/')) {
-          resolvedImportUrl = './' + resolvedImportUrl;
-        }
-        return resolvedImportUrl;
-      });
-      this.output[outLoc] = resolvedCode;
-    }
-    return isSuccess;
-  }
-
-  async writeToDisk() {
-    mkdirp.sync(this.outDir);
-    for (const [outLoc, code] of Object.entries(this.output)) {
-      const encoding = typeof code === 'string' ? 'utf8' : undefined;
-      await fs.writeFile(outLoc, code, encoding);
-    }
-  }
-
-  async getProxy(originalFileLoc: string) {
-    const proxiedCode = this.output[originalFileLoc];
-    const proxiedUrl = originalFileLoc
-      .substr(this.config.buildOptions.out.length)
-      .replace(/\\/g, '/');
-    return wrapImportProxy({
-      url: proxiedUrl,
-      code: proxiedCode,
-      hmr: false,
-      config: this.config,
-    });
-  }
-
-  async writeProxyToDisk(originalFileLoc: string) {
-    const proxyCode = await this.getProxy(originalFileLoc);
-    const importProxyFileLoc = originalFileLoc + '.proxy.js';
-    await fs.writeFile(importProxyFileLoc, proxyCode, 'utf8');
-  }
-}
-
 export async function build(commandOptions: CommandOptions): Promise<SnowpackBuildResult> {
   const {config} = commandOptions;
   const isDev = !!config.buildOptions.watch;
   const isSSR = !!config.buildOptions.ssr;
-
-  // Fill in any command-specific plugin methods.
-  // NOTE: markChanged only needed during dev, but may not be true for all.
-  if (isDev) {
-    for (const p of config.plugins) {
-      p.markChanged = (fileLoc) => onWatchEvent(fileLoc) || undefined;
-    }
-  }
+  const isHMR = getIsHmrEnabled(config);
+  config.buildOptions.resolveProxyImports = !config.optimize?.bundle;
```

Comment:
This line seems to break image import for bundled builds.
[BUG] sowpack bundle build generate import statement for images and json files

If we turn on optimize.bundle, resolveProxyImports is always going to be false, cause the js bundle to import image files directly:

import logo from './logo.svg''

This can be reproduced with @snowpack/app-template-react by turning on optimized.bundle.